collector/uptrack.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package collector
+
+import (
+	"flag"
+	"io/ioutil"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	uptrackUpgradePlanPath = flag.String(
+		"collector.uptrack.upgrade_plan_path",
+		uptrackDefaultUpgradePlanPath(),
+		"Path where Uptrack stores its upgrade_plan YAML file.")
+)
+
+type uptrackCollector struct {
+	timeDesc      *prometheus.Desc
+	planCountDesc *prometheus.Desc
+}
+
+func init() {
+	Factories["uptrack"] = NewUptrackCollector
+}
+
+// The location of the upgrade_plan file is based on the currently running kernel.
+func uptrackDefaultUpgradePlanPath() string {
+	var uname syscall.Utsname
+	syscall.Uname(&uname)
+	return path.Join(
+		"/var/cache/uptrack",
```

Comment:
Monitoring Puppet deployments sounds useful, but again something that IMO shouldn't be in the node-exporter. It's a bit too specialized I'd say and I imaging it being quite deployment specific. It might make sense to have dedicated exporter.
But personally I usually just have my config management system write a textfile collector file with last_run timestamp to monitor deployments.